src/irmin-pack/pack_key.ml

```diff
@@ -61,23 +61,19 @@ let t (type hash) (hash_t : hash Irmin.Type.t) =
       | Dynamic _ | Unknown ->
           Fmt.failwith "Hash must have a fixed-width binary encoding"
   end in
+  (* Equality and ordering on keys respects {i structural} equality semantics,
+     meaning two objects (containing keys) are considered equal even if their
+     children are stored at different offsets (either as duplicates in the same
+     pack file, or inside different pack files), or with different lengths (in
+     the event that the encoding environments were different). *)
```

Comment:
how can the same object have two different encodings? is it due to changing the version of an object (from v0 to v1)?

Reply:
I see, thanks! (using the hash for equality should anyway be enough)